src/xml.rs

```diff
@@ -27,221 +33,429 @@ pub struct Document {
+            rgb_color => Self::Custom(rgb_color),
+        }
+    }
+}
+
+impl<'a> TryFrom<&'a str> for Color {
+    type Error = &'a str;
+
+    fn try_from(value: &'a str) -> Result<Self, Self::Error> {
+        match value {
+            "Rose" => Ok(Self::Rose),
+            "Red" => Ok(Self::Red),
+            "Orange" => Ok(Self::Orange),
+            "Lemon" => Ok(Self::Lemon),
+            "Green" => Ok(Self::Green),
+            "Turquoise" => Ok(Self::Turquoise),
+            "Blue" => Ok(Self::Blue),
+            "Violet" => Ok(Self::Violet),
+            color_str => Color::from_hex(color_str)
+                .map(Color::Custom)
+                .ok_or(value.as_ref()),
+        }
+    }
+}
+
+impl fmt::Display for Color {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Rose => write!(f, "Rose"),
+            Self::Red => write!(f, "Red"),
+            Self::Orange => write!(f, "Orange"),
+            Self::Lemon => write!(f, "Lemon"),
+            Self::Green => write!(f, "Green"),
+            Self::Turquoise => write!(f, "Turquoise"),
+            Self::Blue => write!(f, "Blue"),
+            Self::Violet => write!(f, "Violet"),
+            Self::Custom(rgb) => write!(f, "#{:02X}{:02X}{:02X}", rgb.r, rgb.g, rgb.b),
+        }
+    }
+}
+
+impl Serialize for Color {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.to_string().serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for Color {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct ColorVisitor;
+        impl serde::de::Visitor<'_> for ColorVisitor {
+            type Value = Color;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                formatter.write_str("a color name or hex code")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                Color::try_from(value)
+                    .map_err(|_| E::invalid_value(serde::de::Unexpected::Str(value), &self))
+            }
+        }
+        deserializer.deserialize_str(ColorVisitor)
+    }
 }
 
+/// Star Rating
+///
 /// 0 star = "@0", 1 star = "51", 2 stars = "102", 3 stars = "153", 4 stars = "204", 5 stars = "255"
-#[derive(Debug, PartialEq, Clone, Copy, Serialize, Deserialize)]
-enum StarRating {
-    Zero,
-    One,
-    Two,
-    Three,
-    Four,
-    Five,
-    Unknown(i32),
+#[derive(Debug, PartialEq, Clone, Copy, Serialize_repr, Deserialize_repr)]
+#[repr(u8)]
+pub enum StarRating {
+    /// Zero Stars
+    Zero = 0x00,
+    /// One Star
+    One = 0x33,
+    /// Two Stars
+    Two = 0x66,
+    /// Three Stars
+    Three = 0x99,
+    /// Four Stars
+    Four = 0xCC,
+    /// Five Stars
+    Five = 0xFF,
 }
```

Comment:
since this is a `u8` under the hood, do fractional stars also exist (eg `230` corresponding to roughly `4.5` stars)? Do those need to be modeled here?

Reply:
I think Rekordbox UI doesn't allow this. We could of course try to make a `Custom(u8)` variant, but I don't think it's needed. If it turns out to be necessary, we'll bump the version :)